[DLSP21] Update 03-3.md

jaylizzhn opened the pull request.

Atcold commented:
Ouch… This lesson is already transcribed (see last year).
All it was necessary was the first 5 to 10 minutes.

jaylizzhn commented:
Hi, I have added the part for the first 5~10 minutes(introduction of Diagram.net). Have a look!
Feel free to remove any redundant content.

Atcold commented on docs/en/week03/03-3.md:
```diff
@@ -0,0 +1,110 @@
+
+In the half above, we feed the $x$ to a given encoder to get a prediction $\bar y$. This is called forward propagation.
+
+In the half below, we want to get the prediction $\bar x$ given observation $x$ and $y$. We keep doing gradient descent to make the network output as close as to $y$. This is called amortizing inference.
```
We don't have $x$, right? We want to infer $\bar{x}$ given $y$.
Also, please replace every `x` and `y` with `\vect{x}` and `\vect{y}`.

jaylizzhn replied:
I usually use \vec{x} in latex. I just want to confirm here we use \vect{x} instead of \vec{x}, is that correct?

Atcold replied:
https://github.com/Atcold/NYU-DLSP21/blob/master/docs/_layouts/custom.html#L7

jaylizzhn replied:
fixed 👍

Atcold commented on docs/en/week03/03-3.md:
```diff
@@ -0,0 +1,110 @@
+
+In the half below, we want to get the prediction $\bar x$ given observation $x$ and $y$. We keep doing gradient descent to make the network output as close as to $y$. This is called amortizing inference.
+
+## Spiral classification
```
For this, please, remove the duplicate content (text and figures) and add a link to [last year's writeup](https://atcold.github.io/pytorch-Deep-Learning/en/week02/02-3/).
Sorry for the double work.

Atcold replied:
Feel free to add any clarification you think it's missing from last year's version.

Atcold reviewed (changes requested):
Almost there. A few more edits and we're done here.

Atcold reviewed (changes requested):
Add everything until https://youtu.be/EyKiYVwrdjE?t=754
Remove all images that are not used.

jaylizzhn commented:
Fixed, I have removed all unused images and introduced that backpropagation can be used for inference. Is there anything else you would like me to add?

Atcold commented:
> Add everything until https://youtu.be/EyKiYVwrdjE?t=754

Atcold commented on docs/en/week03/03-3.md:
```diff
@@ -0,0 +1,43 @@
+
+In this example, $\vect{x}$ and $\vect{y}$  are observations.
+
+In the half above, we feed the $\vect{x}$ to a given encoder to get a prediction $\vect {\bar{y}}$. This is called forward propagation.
```
`\bar{\vect{}}`, not the other way around.

Atcold commented:
Please, add all the new content.

jaylizzhn commented:
Sorry, I don't quite understand what you are specifically referring to. 
I thought I had already added all the new content(5~10th minutes).

Atcold commented:
Sorry if I was unclear.

> Add everything until https://youtu.be/EyKiYVwrdjE?t=754

jaylizzhn commented:
From 5:00 to 12:00 you covered the diagram topic. Then 12:00 to 12:34 you talked about the Twitter topic. I still don't understand what more I should add.

Atcold commented:
Everything from 0:00 to 12:34.

jaylizzhn commented:
Yes, I get it wrong sorry. I will add the first 5 minutes soon.

Atcold commented:
No worries. I understand I was not clear. It's my bad, actually.
Yeah, there should be 3 heading, I guess, about the toolings.

The practicum content is mostly the same as last year.
You can say (instead of converting the title into a link), the following content is mostly the same, so here you can find a link, or something along these lines.

jaylizzhn commented:
Finished, have a look. By the way, I converted the title for Typora and Notion to their official site links. If you don't like it, I can remove the links.

Atcold commented on docs/en/week03/03-3.md:
```diff
@@ -0,0 +1,54 @@
+
+
+## Spiral Classification
+The following content is mostly the same, so here you can find what you need(https://atcold.github.io/pytorch-Deep-Learning/en/week02/02-3/)
```
You forgot the square brackets for the link.

Atcold commented on docs/en/week03/03-3.md:
```diff
@@ -0,0 +1,54 @@
+---                    
+
+## [Typora](https://typora.io/)
+Typora is a useful tool to write markdown and latex. It is convenient to write paper and homework, and generating pdf file with Typora.
```
```suggestion
Typora is a useful tool to write markdown with the addition of formulae in LaTeX. It is convenient to write paper and homework, and generating pdf file with Typora.
```

Atcold commented on docs/en/week03/03-3.md:
```diff
@@ -0,0 +1,54 @@
+
+## [Notion](https://www.notion.so/)
+<center>
+<img src="{{site.baseurl}}/images/week03/03-3/figure1.png" style="background-color:#DCDCDC;" /><br>
```
Linked databases is the feature I was trying to stress about. So, perhaps we should use the following instead?
![image](https://user-images.githubusercontent.com/2119355/118657997-f90acf00-b7b9-11eb-88d0-3668d3de967b.png)

I think linking to [this tutorial](https://www.notion.so/Intro-to-databases-fd8cd2d212f74c50954c11086d85997e) would be beneficial. But we need to explain how this should be used to consume online content for keeping up with Deep Learning. I hope you can see my point.

A similar database can be used to keep a reading list for articles one wants to read with its summaries. See work done by [@y0b1byte](https://twitter.com/y0b1byte), namely [this](https://twitter.com/y0b1byte/status/1394394401489707011).

Atcold commented on docs/en/week03/03-3.md:
```diff
@@ -0,0 +1,54 @@
+Here you can place all your favorite stuff. This includes but is not limited to recipes, music, books, notes. Everything in one place, simple and powerful.
+
+
+## Diagram.net
+
+Diagrams.net is a great tool to draw neural network diagrams. Next we will introduce a few rules to make our diagrams more consistent with the ones in lecture.
```
Missing link?

jaylizzhn replied:
I have added more details. Have a look! 
Thanks.

jaylizzhn commented:
I have added more details. Have a look!
Thanks.

Atcold commented:
Thanks!
